File: k9mail/src/main/java/com/fsck/k9/message/ComposePgpEnableByDefaultDecider.java
package com.fsck.k9.message;


import java.util.List;

import com.fsck.k9.crypto.MessageCryptoStructureDetector;
import com.fsck.k9.mail.Message;
import com.fsck.k9.mail.Part;


public class ComposePgpEnableByDefaultDecider {
    public boolean shouldEncryptByDefault(Message localMessage) {
        return messageIsEncrypted(localMessage);
    }

    private boolean messageIsEncrypted(Message localMessage) {
        List<Part> encryptedParts = MessageCryptoStructureDetector.findMultipartEncryptedParts(localMessage);
        return !encryptedParts.isEmpty();
    }
}

File: legacy/core/src/main/java/com/fsck/k9/message/ComposePgpEnableByDefaultDecider.kt
package com.fsck.k9.message

import com.fsck.k9.crypto.MessageCryptoStructureDetector
import com.fsck.k9.mail.Message

class ComposePgpEnableByDefaultDecider {
    fun shouldEncryptByDefault(localMessage: Message?) = messageIsEncrypted(localMessage)

    private fun messageIsEncrypted(localMessage: Message?) =
        localMessage?.let {
            MessageCryptoStructureDetector.findMultipartEncryptedParts(it).isNotEmpty()
        } ?: false
}
